### PR TITLE
Check if output dir exists prior execution

### LINKED ISSUE #125: Saving plots to not-existing directory fails miserable
```
# grzanka at dell in ~/PycharmProjects/pymchelper on git:master ✖︎ [19:21:40]
→ ls -al out2
ls: cannot access out2: No such file or directory

# grzanka at dell in ~/PycharmProjects/pymchelper on git:master ✖︎ [19:21:42]
→ PYTHONPATH=. python3 pymchelper/run.py --many "data/output/dose2009*.bdo" out2 --converter image
INFO:pymchelper.readers.shieldhit:Reading header: data/output/dose20090.bdo
INFO:pymchelper.readers.shieldhit:Reading data: data/output/dose20090.bdo
INFO:pymchelper.readers.shieldhit:Reading header: data/output/dose20091.bdo
INFO:pymchelper.readers.shieldhit:Reading data: data/output/dose20091.bdo
INFO:pymchelper.readers.shieldhit:Reading header: data/output/dose20092.bdo
INFO:pymchelper.readers.shieldhit:Reading data: data/output/dose20092.bdo
INFO:pymchelper.readers.shieldhit:Reading header: data/output/dose20093.bdo
INFO:pymchelper.readers.shieldhit:Reading data: data/output/dose20093.bdo
INFO:pymchelper.readers.shieldhit:Reading header: data/output/dose20094.bdo
INFO:pymchelper.readers.shieldhit:Reading data: data/output/dose20094.bdo
INFO:pymchelper.readers.shieldhit:Reading header: data/output/dose20095.bdo
INFO:pymchelper.readers.shieldhit:Reading data: data/output/dose20095.bdo
INFO:pymchelper.readers.shieldhit:Reading header: data/output/dose20096.bdo
INFO:pymchelper.readers.shieldhit:Reading data: data/output/dose20096.bdo
INFO:pymchelper.readers.shieldhit:Reading header: data/output/dose20097.bdo
INFO:pymchelper.readers.shieldhit:Reading data: data/output/dose20097.bdo
INFO:pymchelper.readers.shieldhit:Reading header: data/output/dose20098.bdo
INFO:pymchelper.readers.shieldhit:Reading data: data/output/dose20098.bdo
INFO:pymchelper.readers.shieldhit:Reading header: data/output/dose20099.bdo
INFO:pymchelper.readers.shieldhit:Reading data: data/output/dose20099.bdo
INFO:pymchelper.writers.plots:Writing: out2/dose2.txt.png
Traceback (most recent call last):
  File "pymchelper/run.py", line 76, in <module>
    sys.exit(main(sys.argv[1:]))
  File "pymchelper/run.py", line 67, in main
    parsed_args.nscale, ErrorEstimate[parsed_args.error])
  File "/home/grzanka/PycharmProjects/pymchelper/pymchelper/detector.py", line 372, in merge_many
    merge_list(group_with_same_core, output_file, conv_names, nan, colormap, nscale, error_estimate)
  File "/home/grzanka/PycharmProjects/pymchelper/pymchelper/detector.py", line 330, in merge_list
    first.save(output_file, conv_names, colormap)
  File "/home/grzanka/PycharmProjects/pymchelper/pymchelper/detector.py", line 147, in save
    writer.write(self)
  File "/home/grzanka/PycharmProjects/pymchelper/pymchelper/writers/plots.py", line 200, in write
    self._save_2d_error_plot(detector, xlist, ylist, elist)
  File "/home/grzanka/PycharmProjects/pymchelper/pymchelper/writers/plots.py", line 147, in _save_2d_error_plot
    plt.savefig(base_name + "_error.png")
  File "/usr/local/lib/python3.4/dist-packages/matplotlib/pyplot.py", line 696, in savefig
    res = fig.savefig(*args, **kwargs)
  File "/usr/local/lib/python3.4/dist-packages/matplotlib/figure.py", line 1563, in savefig
    self.canvas.print_figure(*args, **kwargs)
  File "/usr/local/lib/python3.4/dist-packages/matplotlib/backend_bases.py", line 2232, in print_figure
    **kwargs)
  File "/usr/local/lib/python3.4/dist-packages/matplotlib/backends/backend_agg.py", line 532, in print_png
    filename_or_obj = open(filename_or_obj, 'wb')
FileNotFoundError: [Errno 2] No such file or directory: 'out2/dose2.txt_error.png'
```

### PR DESCRIPTION
**Pull request by nbassler**
Closes #125 ?

### CODE REVIEW

File: pymchelper/run.py [hunk 1]
```diff
@@ -57,6 +58,11 @@ def main(args=sys.argv[1:]):
 
     set_logger_level(parsed_args)
 
+    # check if output directory exists
+    output_dir = os.path.dirname(parsed_args.output)
+    if not(os.path.exists(output_dir)):
+        raise IOError("Directory {}/ does not exist.".format(output_dir))
```
Comment (grzanka) [inline pymchelper/run.py]: Output might also be a filename, for example:
`convertmc NBDoseDD0001.bdo nice_plot.png --converter image `

then `parsed_args.output` is equal to nice_plot.png and it is expected that this file might not exists.

File: pymchelper/run.py [hunk 2]
```diff
@@ -57,6 +58,11 @@ def main(args=sys.argv[1:]):
 
     set_logger_level(parsed_args)
 
+    # check if output directory exists
+    output_dir = os.path.dirname(parsed_args.output)
+    if not(os.path.exists(output_dir)):
```
Comment (grzanka) [inline pymchelper/run.py]: should be:
`if not os.path.exists(...)`

File: pymchelper/run.py [hunk 3]
```diff
@@ -57,6 +58,11 @@ def main(args=sys.argv[1:]):
 
     set_logger_level(parsed_args)
 
+    # check if output directory exists
+    output_dir = os.path.dirname(parsed_args.output)
```
Comment (grzanka) [inline pymchelper/run.py]: Output dir can also be None. It is also expected. In such case writers will automatically construct output filename.
I know, it should be documented.

File: pymchelper/run.py (hunk 1 above)
Comment (nbassler) [inline pymchelper/run.py]: thats why id extract the dirname from the full path.

**Pull request merged**